core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -189,17 +284,18 @@ public synchronized void connect() throws AlluxioStatusException {
             getServiceName(), retryPolicy.getAttemptCount(), e.toString());
         continue;
       }
-      LOG.info("Alluxio client (version {}) is trying to connect with {} @ {}",
-          RuntimeConstants.VERSION, getServiceName(), mAddress);
-
-      mProtocol = ThriftUtils.createThriftProtocol(
-          mTransportProvider.getClientTransport(mParentSubject, mAddress), getServiceName());
+      // Bootstrap once
+      if (HANDSHAKE_NEEDED && !sHandshakeComplete.get()) {
```

Comment:
How does this work when there are multiple clients in a JVM, like block master client, fsm client etc? All of those clients will run the handshake protocol?

Also, worker clients to the master also run the handshake?

Reply:
- though there can be different clients, they will only handshake once as `AbstractClient` is their super class.
- all the clients will handshake to load cluster default. We can have an offline conversation and see if that makes sense.